base.js

```diff
@@ -25,26 +28,65 @@ module.exports.LengthResult = LengthResult;
 module.exports.WriteResult = WriteResult;
 module.exports.ReadResult = ReadResult;
 
-function BufferRW(byteLength, readFrom, writeInto) {
+function BufferRW(poolByteLength, poolReadFrom, poolWriteInto) {
     if (!(this instanceof BufferRW)) {
-        return new BufferRW(byteLength, readFrom, writeInto);
+        return new BufferRW(poolByteLength, poolReadFrom, poolWriteInto);
     }
-    if (typeof byteLength === 'function') this.byteLength = byteLength;
-    if (typeof readFrom === 'function') this.readFrom = readFrom;
-    if (typeof writeInto === 'function') this.writeInto = writeInto;
+
+    if (typeof poolByteLength === 'function') this.poolByteLength = poolByteLength;
+    if (typeof poolReadFrom === 'function') this.poolReadFrom = poolReadFrom;
+    if (typeof poolWriteInto === 'function') this.poolWriteInto = poolWriteInto;
+
+    assert(typeof this.poolByteLength === 'function', 'rw must implement `poolByteLength`');
+    assert(typeof this.poolReadFrom === 'function', 'rw must implement `poolReadFrom`');
+    assert(typeof this.poolWriteInto === 'function', 'rw must implement `poolWriteInto`');
```

Comment:
The above can be simplified by asserting the arguments existence then assigning unconditionally.